client/blocks/stats-navigation/index.js

```diff
@@ -61,6 +62,150 @@ function withNoticeHook( HookedComponent ) {
 	};
 }
 
+const TabsBeforeNavigationImprovements = ( {
+	validNavItems,
+	label,
+	interval,
+	slugPath,
+	selectedItem,
+	showLock,
+	isLegacy,
+	showIntervals,
+	pathTemplate,
+} ) => {
+	return (
+		<>
+			<SectionNav selectedText={ label }>
+				<NavTabs selectedText={ label }>
+					{ validNavItems.map( ( item ) => {
+						const navItem = navItems[ item ];
+						const intervalPath = navItem.showIntervals ? `/${ interval || 'day' }` : '';
+						const itemPath = `${ navItem.path }${ intervalPath }${ slugPath }`;
+						const className = 'stats-navigation__' + item;
+						if ( item === 'store' && config.isEnabled( 'is_running_in_jetpack_site' ) ) {
+							return (
+								<NavItem
+									className={ className }
+									key={ item }
+									onClick={ () =>
+										( window.location.href = `${ this.props.adminUrl }admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` )
+									}
+									selected={ false }
+								>
+									{ navItem.label }
+								</NavItem>
+							);
+						}
+						return (
+							<NavItem
+								className={ className }
+								key={ item }
+								path={ itemPath }
+								selected={ selectedItem === item }
+							>
+								{ navItem.label }
+								{ navItem.paywall && showLock && ' 🔒' }
+							</NavItem>
+						);
+					} ) }
+				</NavTabs>
+
+				{ isLegacy && showIntervals && (
+					<Intervals selected={ interval } pathTemplate={ pathTemplate } />
+				) }
+			</SectionNav>
+
+			{ isLegacy && showIntervals && (
+				<Intervals selected={ interval } pathTemplate={ pathTemplate } standalone />
+			) }
+		</>
+	);
+};
+
+const SelectNav = ( {
```

Comment:
To retain the same effect as before, I suggest using `TabsBeforeNavigationImprovements` for both legacy and improved mobile navigation.

Reply:
Addressed in [b700480](https://github.com/Automattic/wp-calypso/pull/103325/commits/b70048096dabf048271f9d0b22781510aad8e8a4).

However, we can discuss deprecating the `Intervals` component from StatsNavigation in the future. cc @kangzj